platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PreConsensusEventFileManager.java

```diff
@@ -259,79 +274,145 @@ private static void fileSanityChecks(
-     *                                 that may contain events with a generation greater or equal to this value. A value
-     *                                 of {@link #NO_MINIMUM_GENERATION} will cause the returned iterator to walk over
-     *                                 all available event files.
-     * @param requireMinimumGeneration if true, then throw if data for the requested minimum generation is not available
-     *                                 in the event files. If false, then the iterator will walk over events starting at
-     *                                 the first generation on disk that is greater or equal to the requested minimum
-     *                                 generation.
+     * @param minimumGeneration  the desired minimum generation, iterator is guaranteed to walk over all files that may
+     *                           contain events with a generation greater or equal to this value. A value of
+     *                           {@link #NO_MINIMUM_GENERATION} will cause the returned iterator to walk over all
+     *                           available event files.
+     * @param fixDiscontinuities if true, any discontinuities after the requested minimum generation will be "fixed" by
+     *                           deleting all data following the first discontinuity.
      * @return an iterator that walks over event files in order
      */
     public @NonNull Iterator<PreConsensusEventFile> getFileIterator(
-            final long minimumGeneration, final boolean requireMinimumGeneration) {
+            final long minimumGeneration, final boolean fixDiscontinuities) {
 
+        // Edge case: we want all events regardless of generation
+        if (minimumGeneration == NO_MINIMUM_GENERATION) {
+            if (fixDiscontinuities) {
+                scanForDiscontinuities(0);
+            }
+            return files.iterator();
+        }
+
+        // Edge case: there are no files
         if (files.size() == 0) {
-            if (requireMinimumGeneration) {
-                throw new IllegalStateException("No event files are available, cannot iterate over events");
-            } else {
-                return Collections.emptyIterator();
+            logger.warn(STARTUP.getMarker(), "No preconsensus event files available");
+            return Collections.emptyIterator();
+        }
+
+        // Edge case: our first file comes after the requested starting generation
+        if (files.getFirst().getMinimumGeneration() >= minimumGeneration) {
+            // Unless we observe at least one file with a minimum generation less than the requested minimum,
+            // then we can't know for certain that we have all data for the requested minimum generation.
+            logger.warn(
+                    STARTUP.getMarker(),
+                    "The preconscious event stream has insufficient data to guarantee that all events with the "
```

Comment:
"preconscious" implies that the event streams may eventually become conscious, which I think we should do our best to prevent

Reply:
We're operating at the speed of the autonomic nervous system.  No time for consciousness to be involved.

Reply:
Good catch, we wouldn't accidentally want to create a skynet situation.